bouncer/src/repo/manipulator/modelconvertor/import/repo_model_import_3drepo.cpp

```diff
@@ -132,25 +137,76 @@ repo::core::model::MaterialNode *RepoModelImport::parseMaterial(const boost::pro
 	materials.insert(materialNode);
 	matNodeList.push_back(materialNode);
 
-	return materialNode;
+	if(textureId >= 0)
+	{
+		textureIdToParents[textureId].push_back(materialNode->getSharedID());
+	}
+	return;
+}
+
+void RepoModelImport::parseTexture(
+	const boost::property_tree::ptree& textureTree, 
+	char * const dataBuffer)
+{
+	bool fileNameOk	 = textureTree.find("filename") != textureTree.not_found();
+	bool byteCountOK = textureTree.find("numImageBytes") != textureTree.not_found();
+	bool widthOk	 = textureTree.find("width") != textureTree.not_found();
+	bool heightOk	 = textureTree.find("height") != textureTree.not_found();
+	bool idOk		 = textureTree.find("id") != textureTree.not_found();
+
+	if (!byteCountOK ||
+		!widthOk	 ||
+		!heightOk	 ||
+		!idOk)
+	{
+		repoError << "Required texture field missing. Skipping this texture.";
+		missingTextures = true;
+		return;
+	}
+
+	std::string name = fileNameOk ? textureTree.get_child("filename").data() : "";
+	uint32_t byteCount = textureTree.get<uint32_t>("numImageBytes");
+	uint32_t width = textureTree.get<uint32_t>("width");
+	uint32_t height = textureTree.get<uint32_t>("height");
+	uint32_t id = textureTree.get<uint32_t>("id");
+
+	std::vector<uint32_t> DataStartEnd = as_vector<uint32_t>(textureTree, "imageBytes");
```

Comment:
use `uint64_t`

This number can overflow on 32bit if the geometry buffer is > 2gb (which does happen sometimes)

Reply:
nice catch, updated the the data type. The size to be wary of would be 4b though no (uint32 max)? ( kinda like how NTFS systems can only address a file under 4 GB right?)

Reply:
true, was thinking int32_t!